public/app/plugins/datasource/cloudwatch/tracking.test.ts

```diff
@@ -0,0 +1,767 @@
+import { SQLQuery } from '@grafana/aws-sdk';
+import { DashboardLoadedEvent } from '@grafana/data';
+import { reportInteraction } from '@grafana/runtime';
+
+import { CloudWatchQuery } from './types';
+
+import './module';
+
+jest.mock('@grafana/runtime', () => {
+  return {
+    ...jest.requireActual('@grafana/runtime'),
+    reportInteraction: jest.fn(),
+    getAppEvents: () => ({
+      subscribe: jest.fn((e, handler) => {
+        // Trigger test event
+        handler(
```

Comment:
Yep totally agree! Refactored so that the event is now triggered from inside the test. Should make it easy to add new tests if we want to do that in the future. Also moved test event to __mocks__ folder.